File: 3280-convert-date-to-binary/3280-convert-date-to-binary.js
/**
 * @param {string} date
 * @return {string}
 */
var convertDateToBinary = function (date) {
    let strings = date.split("-");
    let ans = "";
    for (let i = 0; i < strings.length; i++) {
        let binary = "";
        let num = parseInt(strings[i]);
        while (num > 0) {
            binary = (num % 2) + binary;
            num = Math.floor(num / 2);
        }
        if (i < strings.length - 1) {
            ans += binary + "-";
        } else {
            ans += binary;
        }
    }
    return ans;
};

File: 3280-convert-date-to-binary/3280-convert-date-to-binary.java
class Solution {
  public String convertDateToBinary(String date) {
    String[] split = date.split("-");
    StringBuilder sb = new StringBuilder();

    for (int i = 0; i < split.length; i++) {
      String binary = Integer.toBinaryString(Integer.parseInt(split[i]));
      sb.append(binary);
      if (i != split.length - 1) {
        sb.append("-");
      }
    }

    return sb.toString();
  }
}